### PR TITLE
Accessibility: Use H2 tag in Lab2 PanelContainer

### PR DESCRIPTION
**Pull request by molly-moen**
Our VPAT popped up errors in AI Chat, Python Lab and Music Lab because the Lab2 PanelContainer heading text is not surrounded by a heading tag. These all use the same component and therefore could be covered by one PR!

To fix this, I updated the central header text to use an H2 element. I only made the central text an h2 because that is where we always put the main header content (i.e. "Instructions", "Workspace", etc.). However, the component does not enforce this. The left and right content is usually buttons. I figured it made more sense to make the component style the heading once rather than have each usage need to specify and style the heading correctly.

As far as I can tell there is no visual difference here. Only Python Lab has an eyes test, but that test showed no changes with this update.

## Screenshots
These look unchanged to me, and the python lab one passed the eyes test, but here are the 3 labs after the change:

Screenshot 2024-12-17 at 3 16 42 PM
Screenshot 2024-12-17 at 3 16 34 PM
Screenshot 2024-12-17 at 3 16 22 PM

## Links
- jira tickets: [A11Y-886 (Music Lab)](https://codedotorg.atlassian.net/browse/A11Y-866), [A11Y-883 (Python Lab)](https://codedotorg.atlassian.net/browse/A11Y-883), [A11Y-1069 (AI Chat)](https://codedotorg.atlassian.net/browse/A11Y-1069)

## Testing story
Tested locally that the headings looked unchanged, and that the Python Lab eyes test still passes.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lab2/views/components/panelContainer.module.scss [hunk 1]
```diff
@@ -31,10 +31,21 @@ $default-spacing: 2px;
   border-bottom: 1px $light_gray_200 solid;
 }
 
+.panelContainerHeaderItem-light {
```
Comment (cnbrenci) [inline apps/src/lab2/views/components/panelContainer.module.scss:34]: nit: I think the preference is to use the design system color (ones under [here ](https://github.com/code-dot-org/code-dot-org/blob/staging/shared/css/color.scss#L190-L191)) `$light_black` and `$light_white` in favor of `$neutral_dark` and `$neutral_white`

Comment (cnbrenci) [reply apps/src/lab2/views/components/panelContainer.module.scss:34]: (they are the same color codes)

Comment (molly-moen) [reply apps/src/lab2/views/components/panelContainer.module.scss:34]: updated!

**Review by cnbrenci — APPROVED** [review]
Wow 3 accessibility tickets with 1 little PR 🌟 🤌 🙌🏻

**Pull request merged**